x/ecocredit/spec/06_tests.md

```diff
@@ -281,3 +269,185 @@ If a user tries to cancel credits and the user provides an invalid batch denomin
+- WHEN - user tries to create a buy order
+- THEN - transaction is successful, buy order is processed
+
+If a user tries to create a buy order and the user provides a quantity that amounts to a total cost that is more than their coin balance, then the transaction fails and the buy order is NOT processed.
+
+- GIVEN - user provides a quantity that amounts to a total cost that is more than their coin balance
+- WHEN - user tries to create a buy order
+- THEN - transaction fails, buy order is NOT processed
+
+If a user tries to create a buy order and the user provides a bid price that amounts to a total cost that is less than or equal to their coin balance, then the transaction is successful and the buy order is processed.
+
+- GIVEN - user provides a bid price that amounts to a total cost that is less than or equal to their coin balance
+- WHEN - user tries to create a buy order
+- THEN - transaction is successful, buy order is processed
+
+If a user tries to create a buy order and the user provides a bid price that amounts to a total cost that is more than their coin balance, then the transaction fails and the buy order is NOT processed.
+
+- GIVEN - user provides a bid price that amounts to a total cost that is more than their coin balance
+- WHEN - user tries to create a buy order
+- THEN - transaction fails, buy order is NOT processed
+
+If a user tries to create a buy order and the user provides a bid price using an ask denom that is allowed, then the transaction is successful and the buy order is processed.
+
+- GIVEN - user provides a bid price using an ask denom that is allowed
+- WHEN - user tries to create a buy order
+- THEN - transaction is successful, buy order is processed
+
+If a user tries to create a buy order and the user provides a bid price using an ask denom that is NOT allowed, then the transaction fails and the buy order is NOT processed.
+
+- GIVEN - user provides a bid price using an ask denom that is NOT allowed
+- WHEN - user tries to create a buy order
+- THEN - transaction fails, buy order is NOT processed
+
+## Allow Ask Denom
+
+If a user tries to add an ask denom and the user submits a governance proposal that is approved, then the transaction is successful and the ask denom is added.
+
+- GIVEN - user submits a governance proposal that is approved
+- WHEN - user tries to add an ask denom
+- THEN - transaction is successful, ask denom is added
+
+If a user tries to add an ask denom and the user submits a governance proposal that is NOT approved, then the transaction fails and the ask denom is NOT added.
+
+- GIVEN - user provides a bid price using an ask denom that is NOT allowed
+- WHEN - user tries to add an ask denom
+- THEN - transaction fails, ask denom is NOT added
+
+If a user tries to add an ask denom and the user submits a transaction directly from their account, then the transaction fails and the ask denom is NOT added.
```

Comment:
Let's use more simple language:
```suggestion
User can't add a new _ask denom_ directly, without creating (or linking... let's be more precise here) a gov proposal.
```

Reply:
I think we should stick with the standard format for acceptance tests: https://openclassrooms.com/en/courses/4544611-write-agile-documentation-user-stories-acceptance-tests/4810081-write-acceptance-tests

Reply:
I don't think changing the language is necessary. I think we should follow the standard format for acceptance tests: https://openclassrooms.com/en/courses/4544611-write-agile-documentation-user-stories-acceptance-tests/4810081-write-acceptance-tests

Reply:
if-then .... is just another way of saying the same thing using "given - when - then". 

The article says about the benefit of using "given - when -then" compared to other format, and it mentions the "if - else" for an exercise:

> Let's look at the first example from above:
> >    If the customer tries to withdraw $20 but her account balance is under $20, then do not allow the withdrawal.
>
> Writing that in Given-When-Then format will look like this:
> ...